mllib/src/test/scala/org/apache/spark/mllib/tree/RandomForestSuite.scala

```diff
@@ -196,6 +196,24 @@ class RandomForestSuite extends FunSuite with MLlibTestSparkContext {
       featureSubsetStrategy = "sqrt", seed = 12345)
     EnsembleTestHelper.validateClassifier(model, arr, 1.0)
   }
+
+  test("subsampling rate in RandomForest"){
+    val arr = EnsembleTestHelper.generateOrderedLabeledPoints(5, 20)
+    val rdd = sc.parallelize(arr)
+    val strategy1 = new Strategy(algo = Classification, impurity = Gini, maxDepth = 2,
+      numClasses = 2, categoricalFeaturesInfo = Map.empty[Int, Int],
+      useNodeIdCache = true, subsamplingRate=0.5)
+    val strategy2 = new Strategy(algo = Classification, impurity = Gini, maxDepth = 2,
+      numClasses = 2, categoricalFeaturesInfo = Map.empty[Int, Int],
+      useNodeIdCache = true)
+
+    val rf1 = RandomForest.trainClassifier(rdd, strategy1, numTrees = 3,
+      featureSubsetStrategy = "auto", seed = 123)
+    val rf2 = RandomForest.trainClassifier(rdd, strategy2, numTrees = 3,
+      featureSubsetStrategy = "auto", seed = 123)
+    assert(rf1.totalNumNodes != rf2.totalNumNodes)
```

Comment:
@jkbradley I've thought about this. This does not seem an ideal test. i.e If this condition satisfies, then the forests are not the same, but there might be cases where the number of nodes are still the same but the forests are not equal. Do you have any better ideas?